### PR TITLE
WT-10116 Return 0 timestamp for all_durable upon first open transaction

### PR DESCRIPTION
**Pull request by Sean04**
Querying the all_durable timestamp during the first transaction of a new instance of WT would incorrectly return the commit_timestamp of that transaction. This fixes that scenario and instead returns a 0 timestamp indicating that there is no all_durable timestamp before any transaction has been committed.

### CODE REVIEW

**Review by agorrod — COMMENTED** [review]
LGTM - did you confirm that this is enough to address the original issue reported?

**Comment by Sean04** [comment]
> LGTM - did you confirm that this is enough to address the original issue reported?

The asserts added to timestamp08 check the original scenario as outlined in the ticket. The only difference is that we are returning '0' not '9' as originally thought in the ticket.

File: src/txn/txn_timestamp.c [hunk 1]
```diff
@@ -162,7 +162,15 @@ __txn_global_query_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t *tsp, cons
     WT_STAT_CONN_INCR(session, txn_query_ts);
     WT_RET(__wt_config_gets(session, cfg, "get", &cval));
     if (WT_STRING_MATCH("all_durable", cval.str, cval.len)) {
-        ts = txn_global->has_durable_timestamp ? txn_global->durable_timestamp : 0;
+        /*
```
Comment (bvpvamsikrishna) [inline src/txn/txn_timestamp.c:165]: is there any reason why this is not under the '__wt_readlock'?

Comment (Sean04) [reply src/txn/txn_timestamp.c:165]: That's a good question, the previous check was also outside the readlock so I was just following that logic. I think the readlock is there for walking the concurrent transactions but I'm not sure if it's needed to check if the durable has been set yet. I can't think of any race conditions at the moment.

This line similarly checks the txn_global->has_oldest_timestamp without the txn_global lock. 
https://github.com/wiredtiger/wiredtiger/blob/439cd2fbc72e0a0dcd19e86205feffd5430440c6/src/txn/txn_timestamp.c#L100

Do you think we should move these to inside the lock?

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Comment by Sean04** [comment]
evergreen merge

**Comment by Sean04** [comment]
evergreen merge

**Comment by Sean04** [comment]
Merging as the commit queue failures are unrelated.

**Pull request merged**